### ISSUE TITLE
Not run on Debian

### ISSUE THREAD

**Issue by ForumLiker** [wwmm/pulseeffects#83]
$ pulseeffects 
Traceback (most recent call last):
  File "/usr/local/bin/pulseeffects", line 6, in <module>
    from PulseEffects.application import Application
ImportError: No module named 'PulseEffects'

Debian 9, pulseeffect v 3.0.0, all needs are install

**Comment by AsavarTzeth** [comment]
This happens because `/usr/local` likely is not within your python path, which should be an issue on all distributions as far as I know. I did not consider this when I implemented the new build system. Then again there are still few python applications that use Meson.

Unfortunately I do not have a solution at this time. I will try and find a way, but I cannot promise anything.

**Comment by ForumLiker** [comment]
Good. I'll wait for your positive result))))

**Comment by AsavarTzeth** [comment]
Well I did some searching and in the end my conclusion is what I guessed it would be.

You need to install into a **/usr** prefix, so that the python modules :
- `PulseEffects` 
- `PulseEffectsCalibration`

are installed in: `/usr/lib/python*/site-packages`.

The only alternative is to add `/usr/local/lib/python*/site-packages` to your python path, but I don't feel comfortable enough with that to recommend it.

This is in fact what they developers [Piper](https://github.com/libratbag/piper#installing-piper) are recommending at their project page. I mention it because it is the best comparison to this project in terms of language and build system.

So perhaps I should change the [installation instructions](https://github.com/wwmm/pulseeffects/wiki/Installation-from-Source#build--install) into:

```
meson _build --prefix=/usr
sudo ninja -C _build install
```

The only negative thing with defaulting to this in the instructions is that some less experienced users might report issues about package file conflicts.

Perhaps I should complement this with:

**Uninstalling**
```
sudo ninja -C _build uninstall
```

and a disclaimer or something. What do you think? I tried it on Arch Linux and I think it should work with Debian as well.

**Comment by ForumLiker** [comment]
how to run after 
`meson _build --prefix=/usr
sudo ninja -C _build install`
`# pulseeffects
sudo: pulseeffects: command not found`

**Comment by AsavarTzeth** [comment]
Strange... From this would indicate that you are having issues on Debian that I cannot replicate on Arch Linux. Perhaps I should spin up a virtual machine and test it on Debian 9. I guess it could be a Debian specific issue.

What does output do you get from `python3 -c "import sys; print('\n'.join(sys.path))"`?

And could you add the output from the `sudo ninja -C _build install` command?

**Comment by AsavarTzeth** [comment]
Oh I see! I ran `python3 -c "import sys; print('\n'.join(sys.path))"` myself and this is my output:

```
/usr/lib/python35.zip
/usr/lib/python3.5
/usr/lib/python3.5/plat-x86_64-linux-gnu
/usr/lib/python3.5/lib-dynload
/usr/local/lib/python3.5/dist-packages
/usr/lib/python3/dist-packages
```

I also read this [Debian wiki page](https://wiki.debian.org/Python). It seems `site-packages` is not there by default at all.

So it is very much a Debian specific thing. I guess I got some more reading to do.

**Comment by AsavarTzeth** [comment]
On another topic, how are you dealing with the dependencies? Debian 9 ships some packages that are too old for PulseEffects 3.0.0.

Are you trying to build a PPA or your own Debian repository, which includes all of these dependencies? If not, then the Flatpak version simply is a better choice.

In any case I would be curious what the package names are for the dependencies. I can't seem to get past the dependency checks of Meson.

**Comment by ForumLiker** [comment]
1)
$ python3 -c "import sys; print('\n'.join(sys.path))"

/usr/lib/python35.zip
/usr/lib/python3.5
/usr/lib/python3.5/plat-x86_64-linux-gnu
/usr/lib/python3.5/lib-dynload
/usr/local/lib/python3.5/dist-packages
/usr/lib/python3/dist-packages

2)
~/pulseeffects-3.0.0$ meson build
The Meson build system
Version: 0.42.1
Source dir: /home/kons/pulseeffects-3.0.0
Build dir: /home/kons/pulseeffects-3.0.0/build
Build type: native build
Project name: PulseEffects
Build machine cpu family: x86_64
Build machine cpu: x86_64
Message: Checking dependencies
Found pkg-config: /usr/bin/pkg-config (0.29)
Native dependency gtk+-3.0 found: YES 3.22.11
Native dependency pygobject-3.0 found: YES 3.22.0
Native dependency py3cairo found: YES 1.10.0
Native dependency pangocairo found: YES 1.40.5
Native dependency gstreamer-1.0 found: YES 1.12.3
Native dependency gstreamer-plugins-bad-1.0 found: YES 1.12.3
Message: Getting python install path
Message: Installing PulseEffects
Configuring pulseeffects using configuration
Message: Installing data files
Program meson_post_install.py found: YES (/usr/bin/env python3 /home/kons/pulseeffects-3.0.0/meson_post_install.py)
Build targets in project: 5

 all need dependencies make from source

3)
~/pulseeffects-3.0.0/build$ ninja
[2/2] Generating com.github.wwmm.pulse...pdata.xml_merge with a custom command.
kons@debian:~/pulseeffects-3.0.0/build$ 
4)
~/pulseeffects-3.0.0/build$ sudo ninja install
[sudo] password for kons: 
[0/1] Installing files.
Installing subdir /home/kons/pulseeffects-3.0.0/PulseEffects to /usr/local/lib/python3.5/site-packages
Installing subdir /home/kons/pulseeffects-3.0.0/PulseEffectsCalibration to /usr/local/lib/python3.5/site-packages
Installing data/com.github.wwmm.pulseeffects.appdata.xml to /usr/local/share/metainfo/com.github.wwmm.pulseeffects.appdata.xml
Installing data/com.github.wwmm.pulseeffects.desktop to /usr/local/share/applications/com.github.wwmm.pulseeffects.desktop
Installing /home/kons/pulseeffects-3.0.0/build/pulseeffects to /usr/local/bin
Installing /home/kons/pulseeffects-3.0.0/data/com.github.wwmm.pulseeffects.gschema.xml to /usr/local/share/glib-2.0/schemas
Installing /home/kons/pulseeffects-3.0.0/data/com.github.wwmm.pulseeffects.sinkinputs.gschema.xml to /usr/local/share/glib-2.0/schemas
Installing /home/kons/pulseeffects-3.0.0/data/com.github.wwmm.pulseeffects.sourceoutputs.gschema.xml to /usr/local/share/glib-2.0/schemas
Installing /home/kons/pulseeffects-3.0.0/data/pulseeffects.svg to /usr/local/share/icons/hicolor/scalable/apps
Running custom install script '/usr/bin/python3 /usr/local/bin/meson --internal gettext install --subdir=po --localedir=share/locale --pkgname=PulseEffects'
Installing /home/kons/pulseeffects-3.0.0/build/po/cs.gmo to /usr/local/share/locale/cs/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/de.gmo to /usr/local/share/locale/de/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/fr_FR.gmo to /usr/local/share/locale/fr_FR/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/hr.gmo to /usr/local/share/locale/hr/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/it_IT.gmo to /usr/local/share/locale/it_IT/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/pt_BR.gmo to /usr/local/share/locale/pt_BR/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/sk.gmo to /usr/local/share/locale/sk/LC_MESSAGES/PulseEffects.mo
Installing /home/kons/pulseeffects-3.0.0/build/po/sv.gmo to /usr/local/share/locale/sv/LC_MESSAGES/PulseEffects.mo
Running custom install script '/usr/bin/env python3 /home/kons/pulseeffects-3.0.0/meson_post_install.py'
Compiling gsettings schemas...
Updating icon cache...
Updating desktop database...

**Comment by AsavarTzeth** [comment]
Of course, how stupid of me regarding the dependencies.

I will focus on the original issue then. In any case, there should be a way to install the python modules in the right place on Debian.

**Comment by ForumLiker** [comment]
I need to make a correction in meson.build?

what should meson.build become after editing?

**Comment by AsavarTzeth** [comment]
It is interesting that you mention this. You could test a possible fix I commited yesterday.

Herr is a newer meson.build file. I still need to test it on some distibutions though.

https://github.com/AsavarTzeth/pulseeffects/blob/fix-debian-dist-packages/meson.build

**Comment by ForumLiker** [comment]
~/pulseeffects-3.0.0$ meson build
The Meson build system
Version: 0.42.1
Source dir: /home/kons/pulseeffects-3.0.0
Build dir: /home/kons/pulseeffects-3.0.0/build
Build type: native build

Meson encountered an error in file meson.build, line 1, column 0:
lexer
project('PulseEffects',
^

**Comment by AsavarTzeth** [comment]
I cannot replicate this with Meson 0.42.1 on any distribution with the new file. How did you fetch it?

Did you edit the file manually? If you did I suggest downloading to make sure there is no user error on your end.
- `curl -OL https://raw.githubusercontent.com/AsavarTzeth/pulseeffects/fix-debian-dist-packages/meson.build`
- `wget https://raw.githubusercontent.com/AsavarTzeth/pulseeffects/fix-debian-dist-packages/meson.build`

**Comment by ForumLiker** [comment]
I created it, copied and pasted the content from your link

**Comment by AsavarTzeth** [comment]
That is strange. If you are 100% sure that you did not make a mistake, then I don't know what could be the cause. I managed to get PulseEffects running with this change.

The error message naturally does not tell me much and I could not find much information on it. Thus I am unsure if I can be of much use at this point.

In any case the fix did work in my clean test environments an is now merged. Hopefully, if there are any issues with it that will be soon be noticed by more people.
